### PR TITLE
Fix lag on restart with scaled avatar

### PR DESCRIPTION
**Pull request by HifiExperiments**
This fixes https://github.com/kasenvr/project-athena/issues/48, not https://github.com/kasenvr/project-athena/issues/77

### CODE REVIEW

File: libraries/physics/src/CharacterController.cpp [hunk 1]
```diff
@@ -543,12 +543,8 @@ void CharacterController::setLocalBoundingBox(const glm::vec3& minCorner, const
         _minStepHeight = DEFAULT_MIN_STEP_HEIGHT_FACTOR * (_halfHeight + _radius);
         _maxStepHeight = DEFAULT_MAX_STEP_HEIGHT_FACTOR * (_halfHeight + _radius);
 
-        if (_physicsEngine) {
```
Comment (HifiExperiments) [inline libraries/physics/src/CharacterController.cpp:546]: The issue was that on startup, before the physics engine loaded, we would set the ADD flags repeatedly as we scaled to the correct size without setting the REMOVE flags, so we'd end up with a ton of duplicate motion states.  The number would depend on loading times and the target scale (smaller scales take longer to get to, so the problem is worse).  In one case, I had 1500 motion states when it should have been 50.

**Comment by kasenvr** [comment]
Wait, so this will solve my avatar lagging on load until such a time that I resize?

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR547/Vircadia-Alpha-PR547-1346a15_cbce911e.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR547/Vircadia-Alpha-PR547-1346a15.tar_cbce911e.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR547/Vircadia-Alpha-PR547-1346a15.tar_cbce911e.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR547/Vircadia-Alpha-PR547-1346a15_cbce911e.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR547/Vircadia-Alpha-PR547-1346a15_cbce911e.exe

**Comment by HifiExperiments** [comment]
@kasenvr yeah!

File: interface/src/avatar/MyCharacterController.cpp [hunk 2]
```diff
@@ -407,13 +407,13 @@ void MyCharacterController::clearDetailedMotionStates() {
 }
 
 void MyCharacterController::buildPhysicsTransaction(PhysicsEngine::Transaction& transaction) {
-    for (size_t i = 0; i < _detailedMotionStates.size(); i++) {
```
Comment (daleglass) [inline interface/src/avatar/MyCharacterController.cpp:410]: This doesn't appear to make any functional change, and to just be a cleanup.

Perhaps this should be in a separate PR?

File: interface/src/avatar/OtherAvatar.cpp [hunk 3]
```diff
@@ -116,10 +116,10 @@ void OtherAvatar::updateSpaceProxy(workload::Transaction& transaction) const {
 
 int OtherAvatar::parseDataFromBuffer(const QByteArray& buffer) {
     int32_t bytesRead = Avatar::parseDataFromBuffer(buffer);
-    for (size_t i = 0; i < _detailedMotionStates.size(); i++) {
```
Comment (daleglass) [inline interface/src/avatar/OtherAvatar.cpp:119]: Same here

File: libraries/physics/src/CharacterController.cpp (hunk 1 above)
Comment (daleglass) [reply libraries/physics/src/CharacterController.cpp:546]: Hmm, what could be the original reasoning for the `if (_physicsEngine)`? Is that an attempt at optimization, or could it be protecting something?

File: interface/src/avatar/MyCharacterController.cpp (hunk 2 above)
Comment (HifiExperiments) [reply interface/src/avatar/MyCharacterController.cpp:410]: these are leftover from when I was trying to track down the problem.  since this is a short PR and these changes theoretically improve performance, I would vote to keep them

File: libraries/physics/src/CharacterController.cpp (hunk 1 above)
Comment (HifiExperiments) [reply libraries/physics/src/CharacterController.cpp:546]: my guess is that instead of setting flags, this code used to do actual work here.  I don’t see any way that removing this if could cause problems.  perhaps if we actually added an if around the ADD flags, we could optimize this to do less work on startup, but that seems a lot riskier, since I could imagine it leading to cases where we never create the motion states
